be/src/storage/lake/lake_persistent_index.cpp

```diff
@@ -248,23 +248,50 @@ Status LakePersistentIndex::try_replace(size_t n, const Slice* keys, const Index
     return Status::OK();
 }
 
-std::unique_ptr<sstable::Iterator> LakePersistentIndex::prepare_merging_iterator() {
+Status LakePersistentIndex::prepare_merging_iterator(
+        const TabletMetadata& metadata, TxnLogPB* txn_log,
+        std::vector<std::shared_ptr<PersistentIndexSstable>>* merging_sstables,
+        std::unique_ptr<sstable::Iterator>* merging_iter_ptr) {
     sstable::ReadOptions read_options;
+    // No need to cache input sst's blocks.
+    read_options.fill_cache = false;
     std::vector<sstable::Iterator*> iters;
+    DeferOp free_iters([&] {
+        for (sstable::Iterator* iter : iters) {
+            delete iter;
+        }
+    });
+
     auto max_compaction_versions = config::lake_pk_index_sst_max_compaction_versions;
     iters.reserve(max_compaction_versions);
-    for (const auto& sstable : _sstables) {
-        sstable::Iterator* iter = sstable->new_iterator(read_options);
+    std::stringstream ss_debug;
+    for (const auto& sstable_pb : metadata.sstable_meta().sstables()) {
```

Comment:
is it possible to make a deep copy of _sstables, instead of doing this init all over again from scratch?

Reply:
Makre a deep copy of `_sstables` won't be better, because it will load something we don't need (E.g. filter). And so need to add mutex for `_sstables`, it will Introducing additional burdens

Reply:
wondering what if implements read-write lock in LakePersistentIndex to make it thread safe. the _sstables is only updated in the last minute of major compaction done, other places are all read-only. Can be minimal impact or not?

Reply:
`RandomAccessFile` in `LakePersistentIndex` isn't thread safe.

Reply:
I think I can open new stable for compaction and reuse same block cache with current sstable, which can be minimal impact